### PR TITLE
Add support of variant prop for FilterButton

### PR DESCRIPTION
**Pull request by adguernier**
This PR attempts to add MUI `variant` support for `<FilterButton>`

### CODE REVIEW

File: packages/ra-ui-materialui/src/list/filter/FilterButton.stories.tsx [hunk 1]
```diff
@@ -372,3 +372,45 @@ export const WithAutoCompleteArrayInput = (args: {
         </MemoryRouter>
     );
 };
+
+export const WithMuiButtonProps = () => {
```
Comment (erwanMarmelab) [inline packages/ra-ui-materialui/src/list/filter/FilterButton.stories.tsx:376]: **suggestion:** ‏Could  it be interesting to have 2 stories to show it: `size` and `variant`?

Comment (adguernier) [reply packages/ra-ui-materialui/src/list/filter/FilterButton.stories.tsx]: Done

File: packages/ra-ui-materialui/src/list/filter/FilterButton.stories.tsx [hunk 2]
```diff
@@ -179,17 +182,25 @@ const ListToolbar = (props: {
...
-            <TextField source="id" />
-            <TextField source="title" />
-            <TextField source="body" />
-        </Datagrid>
-        <Pagination />
-    </ListBase>
-);
+    buttonProps?: FilterButtonProps;
+}) => {
+    console.log('buttonPropssss', { ...props.buttonProps });
```
Comment (erwanMarmelab) [inline packages/ra-ui-materialui/src/list/filter/FilterButton.stories.tsx:187]: console.log spotted

File: packages/ra-ui-materialui/src/list/filter/FilterButton.stories.tsx [hunk 3]
```diff
@@ -372,3 +383,65 @@ export const WithAutoCompleteArrayInput = (args: {
         </MemoryRouter>
     );
 };
+
+export const WithButtonVariant = () => {
```
Comment (erwanMarmelab) [inline packages/ra-ui-materialui/src/list/filter/FilterButton.stories.tsx:387]: **polish:** ‏May we name it as the prop name

```suggestion
export const Variant = () => {
```

File: packages/ra-ui-materialui/src/list/filter/FilterButton.stories.tsx [hunk 4]
```diff
@@ -372,3 +383,65 @@ export const WithAutoCompleteArrayInput = (args: {
...
+                            }}
+                        />
+                    }
+                />
+            </Admin>
+        </MemoryRouter>
+    );
+};
+
+export const WithButtonSize = () => {
```
Comment (erwanMarmelab) [inline packages/ra-ui-materialui/src/list/filter/FilterButton.stories.tsx:418]: **polish:** ‏May we name it as the prop name

```suggestion
export const Size = () => {
```

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**